server/src/main/java/io/crate/execution/engine/indexing/ShardedRequests.java

```diff
@@ -81,16 +78,11 @@ public void add(TItem item, long itemSizeInBytes, ShardLocation shardLocation, R
     }
 
     void addFailedItem(String sourceUri, String readFailure, Long lineNumber) {
-        List<ReadFailureAndLineNumber> itemsWithFailure = itemsWithFailureBySourceUri.computeIfAbsent(
-            sourceUri, k -> new ArrayList<>());
+        Set<ReadFailureAndLineNumber> itemsWithFailure = itemsWithFailureBySourceUri.computeIfAbsent(
+            sourceUri, k -> new HashSet<>());
```

Comment:
Why the change here? How can it happen that there are multiple duplicate errors within the same line?

Reply:
That's basically my current issue/question - details in the [comment](https://github.com/crate/crate/pull/12229#discussion_r834345142) above. In debugger` sourceUriFailureInput.value();` holds value of the previous iteration when parsing second line - I just came up with another idea how to "reset input" - would have to use `lineContext.setFailure(null) `after each process

Reply:
https://github.com/crate/crate/pull/12229/commits/af7739f84577ad44b68b35c6268e906da0dbb461

reverted Set and used another approach to avoid side effects of the prev iteration - now duplicate error is resolved but success count is 0 - `setRawSource(null)` in catch block probably wrong